samcli/local/apigw/service.py

```diff
@@ -14,6 +14,18 @@
 LOG = logging.getLogger(__name__)
 
 
+class CaseInsensitiveDict(dict):
```

Comment:
Please add tests for this class. It will help me (and others) understanding the application of this.

Reply:
We should have both unit and functional. We need to make sure we are unit testing everything. The CodeCoverage show branch 22-23 missing and we are not really testing the other paths, it just so happens we execute those paths by creating the object. Ideally, we would mock out this class in the other tests, and then add explicit tests for this class.

Reply:
Added unit tests to ```tests/functional/local/apigw/test_service.py``` you should now have full code coverage of this class.